shared/js/background/events.es6.js

```diff
@@ -41,19 +43,31 @@ async function onInstalled (details) {
         experiment.setActiveExperiment()
     }
 
+    // remove any orphaned session rules (can happen on extension update/restart)
+    if (manifestVersion === 3) {
+        await settings.ready()
+
+        await flushSessionRules()
+    }
+
     // Inject the email content script on all tabs upon installation (not needed on Firefox)
-    if (browserName !== 'moz') {
-        const tabs = await browser.tabs.query({})
-        for (const tab of tabs) {
-            // Ignore URLs that we aren't permitted to access
-            if (tab.url.startsWith('chrome://')) {
-                continue
+    // FIXME the below code throws an unhandled exception in MV3
```

Comment:
Have you filed an issue for this? Cc @jdorweiler

Reply:
@ladamski looks like this is used for email autofill. Can you check that it's still working?

Reply:
We discussed this prior somewhere in the CTL project, trying to find a reference to it (Asana permitting).  It only runs on ext update/install.

Reply:
Regarding sendMessage, I think we're probably due for a bit of a small security pass through the CTL code.  We have multiple messaging channels in different browsers (incl Mac), and they have all changed a bit over time, so I'll file a separate task to test this more thoroughly.

Reply:
👍 for autofill. Can you make a followup to make sure we test it after this.